p3/plot/backend/pgfplots.py

```diff
@@ -0,0 +1,434 @@
+        app_colors = self.__get_colors(applications, app_style.colors)
+        app_colors_rgb = {}
+        for k in app_colors:
+            app_colors_rgb[app_to_tex_name[k]] = str(
+                matplotlib.colors.to_rgb(app_colors[k]),
+            ).strip("()")
+
+        # Build a dictionary of platforms to labels
+        plat_labels = dict(zip(platforms, string.ascii_uppercase))
+
+        # Choose colors for each platform and then convert the dictionary to
+        # RGB colors using the platform labels
+        plat_colors = self.__get_colors(platforms, plat_style.colors)
+        plat_colors_rgb = {}
+        for k in plat_colors:
+            plat_colors_rgb[plat_labels[k]] = str(
+                matplotlib.colors.to_rgb(plat_colors[k]),
+            ).strip("()")
+
+        # Choose markers for each application
+        markers = app_style.markers
+        if not isinstance(markers, (list, tuple)):
+            raise ValueError("Unsupported type provided for app_markers")
+
+        # build a dictionary of app line specifications for TeX
+        app_line_specs = {}
+        for app, mark in zip(applications, markers):
+            app_line_specs[
+                app
+            ] = f"{app_to_tex_name[app]}, thick, solid, mark={mark}"
+
+        # Choose labels for each platform
+        plat_labels = dict(zip(platforms, string.ascii_uppercase))
+
+        # Set the number of rows in the platform key (if set)
+        # NOTE: This is different to matplotlib because PGF plots uses columns,
+        #       and then transposes (so columns become rows)
+        plat_legend_nrows = plat_legend.kwargs["nrows"]
+
+        plotylabel = ""
+        if eff_column == "app eff":
+            plotylabel = "Application Efficiency"
+        elif eff_column == "arch eff":
+            plotylabel = "Architectural Efficiency"
+        else:
+            msg = "Unexpected efficiency column name: %s"
+            raise ValueError(msg % (eff_column))
+
+        # Build an dictionary with "application name, [ data ]" for the
+        # efficiency plot
+        plot_effs = {}
+        for app_name in applications:
+            app_df = df[
+                (df["application"] == app_name) & (df[eff_column] > 0.0)
+            ]
+            app_df = app_df.sort_values(by=[eff_column], ascending=False)
+            yvalues = app_df[eff_column]
+            supported_platforms = list(app_df["platform"])
+            xvalues = np.arange(1, len(supported_platforms) + 1)
+            plot_effs[app_name] = tuple(zip(xvalues, yvalues))
+
+        # Build a dictionary with "application name, (application name, qp)"
+        # for the pp bar plot
+        pp = p3.metrics.pp(df)
+        pp_column = eff_column.replace("eff", "pp")
+        pp_bars = {
+            app: f"({app}, {app_pp})"
+            for app, app_pp in zip(pp["application"], pp[pp_column])
+        }
+
+        # Build a dictionary with "application name, [ A, B, C, etc ]", etc
```

Comment:
It was at least semi-intentional, in that it can be many entries with different application names and then a list of the alpha representations of platforms as it goes. But the second etc is probably not necessary, so I've removed it.